lxc/cluster.go

```diff
@@ -810,3 +815,99 @@ func (c *cmdClusterRevokeToken) Run(cmd *cobra.Command, args []string) error {
 
 	return fmt.Errorf(i18n.G("No cluster join token for member %s on remote: %s"), resource.name, resource.remote)
 }
+
+// Update Certificatess
+type cmdClusterUpdateCertificates struct {
+	global  *cmdGlobal
+	cluster *cmdCluster
+}
+
+func (c *cmdClusterUpdateCertificates) Command() *cobra.Command {
+	cmd := &cobra.Command{}
+	cmd.Use = usage("update-certificate", i18n.G("[<remote>:] <cert.crt> <cert.key>"))
+	cmd.Aliases = []string{"update-cert"}
+	cmd.Short = i18n.G("Update cluster certificates")
+	cmd.Long = cli.FormatSection(i18n.G("Description"),
+		i18n.G("Update cluster certificates with PEM certificate and key read from input files"))
+
+	cmd.RunE = c.Run
+	return cmd
+}
+
+func (c *cmdClusterUpdateCertificates) Run(cmd *cobra.Command, args []string) error {
+	conf := c.global.conf
+
+	exit, err := c.global.CheckArgs(cmd, args, 2, 3)
+	if exit {
+		return err
+	}
+
+	// Parse remote
+	remote := ""
+	certFile := args[0]
+	keyFile := args[1]
+	if len(args) == 3 {
+		remote = args[0]
+		certFile = args[1]
+		keyFile = args[2]
+	}
+
+	resources, err := c.global.ParseServers(remote)
+	if err != nil {
+		return err
+	}
+
+	resource := resources[0]
+
+	// Check if clustered.
+	cluster, _, err := resource.server.GetCluster()
+	if err != nil {
+		return err
+	}
+
+	if !cluster.Enabled {
+		return fmt.Errorf(i18n.G("LXD server isn't part of a cluster"))
+	}
+
+	if !shared.PathExists(certFile) {
+		return fmt.Errorf(i18n.G("Could not find certificate file path: %s"), certFile)
+	}
+
+	if !shared.PathExists(keyFile) {
+		return fmt.Errorf(i18n.G("Could not find certificate key file path: %s"), keyFile)
+	}
+
+	cert, err := ioutil.ReadFile(certFile)
+	if err != nil {
+		return fmt.Errorf(i18n.G("Could not read certificate file: %s with error: $v"), certFile, err)
+	}
+
+	key, err := ioutil.ReadFile(keyFile)
+	if err != nil {
+		return fmt.Errorf(i18n.G("Could not read certificate key file: %s with error: $v"), keyFile, err)
+	}
+
+	certificates := api.ClusterCertificatesPost{
+		ClusterCertificate:    base64.StdEncoding.EncodeToString(cert),
+		ClusterCertificateKey: base64.RawStdEncoding.EncodeToString(key),
+	}
+
+	err = resource.server.UpdateClusterCertificates(certificates)
+	if err != nil {
+		return err
+	}
+
+	if remote != "" {
```

Comment:
Hmm, so that's interesting. I didn't think of us doing this in the first place :)

I'm not opposed to doing this as it is a nice touch for users who aren't using valid certificates and so do have a cert in ServerCertPath.

I'd however change it to something like:
```
// Update existing remote certificate.
certf := conf.ServerCertPath(resource.remote)
if shared.PathExists(certf) {
    err = ioutil.WriteFile(certf, cert, 0644)
    if err != nil {
        return err
    }
}
```

Note that your `if remote != ""` check is something to avoid as it doesn't actually tell you if you're dealing with a remote server or not. All it tells you is whether you're talking to your default remote or not.

Someone doing:
```
lxc remote switch my-cluster
lxc cluster update-certificate mycert.pem mykey.pem
```

May still need an update to `~/.config/lxc/servercerts/my-cluster.crt` as they are still dealing with a remote server.

Reply:
Ah and note the mode, 0644 as this is a file and not a directory :)